gapis/api/vulkan/frame_loop.go

```diff
@@ -258,35 +324,50 @@ func (f *frameLoop) detectChangedResource(ctx context.Context, startState *api.G
 		log.E(ctx, "Mutate error: [%v].", err)
 	}
 
-	st := GetState(f.backupState)
+	apiState := GetState(f.backupState)
+
 	// Find out changed buffers.
-	vkBuffers := st.Buffers().All()
-	for k, buffer := range vkBuffers {
+	for bufferKey, buffer := range apiState.Buffers().All() {
```

Comment:
I think this will be unfortunately non-deterministic. (I think the previous was also non-determinstic) this will create a pain for validating when you change things.